### PR TITLE
Add support for Reference Assemblies packages

### PR DESCRIPTION
**Pull request by tmat**
Implement https://github.com/dotnet/designs/pull/33 in RepoToolset.

The package that's referenced pulls down only the reference assemblies for target frameworks that the project targets.

### CODE REVIEW

**Comment by tmat** [comment]
@dsplaisted PTAL
@jaredpar @weshaggard @chcosta @markwilkie @natemcmaster FYI

File: sdks/RepoToolset/tools/ProjectDefaults.props [hunk 1]
```diff
@@ -132,4 +132,11 @@
     </When>
   </Choose>
 
+  <!-- 
+    Implements proposal https://github.com/dotnet/designs/pull/33.
+  -->
+  <ItemGroup Condition="'$(UsingToolNetFrameworkReferenceAssemblies)' == 'true'">
```
Comment (dsplaisted) [inline sdks/RepoToolset/tools/ProjectDefaults.props]: I'd suggest also conditioning this on whether the `TargetFrameworkIdentifier` is `.NETFramework`.

Comment (tmat) [reply sdks/RepoToolset/tools/ProjectDefaults.props]: Wouldn't they be ignored if not? The build works with just this condition. Is there a potential problem?

Comment (dsplaisted) [reply sdks/RepoToolset/tools/ProjectDefaults.props]: It's not a functional issue, but you'll end up with package references that aren't needed for projects that target something besides .NET Framework.  They won't actually do anything due to the logic inside them, but it seems cleaner not to reference them when you know they won't be needed.

**Pull request merged**